libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -2047,7 +2218,50 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         }
     }
 
-    private fun createChildScope() = threadController.getMainScopeAndRootJob().scope
+    private fun runOnNavigationLooper(block: () -> Unit) {
+        if (navigationOptions.looper == Looper.myLooper()) {
+            block()
+        } else {
+            handler.post {
+                runIfNotDestroyed {
```

Comment:
Why do we only check that it's not destroyed if this branch? Do we assume `runOnNavigationLooper` will only be called inside the `runIfNotDestroyed` block? I think we shouldn't. On the other hand, the method name does not imply we will check `isDestroyed` flag here. I suggest renaming it or dividing into 2 (but if we often use them in a chain, renaming will do).